src/encdatavault_fmt_plug.c

```diff
@@ -0,0 +1,411 @@
+	                             self->params.max_keys_per_crypt, MEM_ALIGN_WORD);
+	cracked = mem_calloc(sizeof(*cracked), self->params.max_keys_per_crypt);
+	cracked_size = sizeof(*cracked) * self->params.max_keys_per_crypt;
+	cracked_count = self->params.max_keys_per_crypt;
+}
+
+static void done(void)
+{
+	MEM_FREE(saved_key);
+	MEM_FREE(cracked);
+}
+
+static int valid(char *ciphertext, struct fmt_main *self)
+{
+	char *p = ciphertext, *ctcopy, *keeptr;
+	int extra;
+
+	if (strncmp(ciphertext, FORMAT_TAG, FORMAT_TAG_LENGTH) != 0)
+		return 0;
+
+	ctcopy = strdup(ciphertext);
+	keeptr = ctcopy;
+	ctcopy += FORMAT_TAG_LENGTH;
+	if ((p = strtokm(ctcopy, "$")) == NULL) // version
+		goto err;
+	if (!isdec(p))
+		goto err;
+	if (atoi(p) != 1 && atoi(p) != 3) {
+		fprintf(stderr, "Warning: version %d not supported!\n", atoi(p));
+		goto err;
+	}
+	if ((p = strtokm(NULL, "$")) == NULL)   // algorithm id
+		goto err;
+	if (!isdec(p))
+		goto err;
+	if (atoi(p) > 4 || atoi(p) < 1) {
+		fprintf(stderr, "Warning: algorithm id %d not supported!\n", atoi(p));
+		goto err;
+	}
+	if ((p = strtokm(NULL, "$")) == NULL)   // Nonce
+		goto err;
+	if (hexlenl(p, &extra) != ENC_IV_PREFIX_SIZE * 2 || extra)
+		goto err;
+	if ((p = strtokm(NULL, "$")) == NULL)   // Encrypted header
+		goto err;
+	if (hexlenl(p, &extra) != ENC_SIG_SIZE * 2 || extra)
+		goto err;
+	MEM_FREE(keeptr);
+	return 1;
+
+err:
+	MEM_FREE(keeptr);
+	return 0;
+}
+
+static void *get_salt(char *ciphertext)
+{
+	static struct custom_salt cs = { 0 };
+	int i;
+	char *p = ciphertext, *ctcopy, *keeptr;
+
+	memset(&cs, 0, sizeof(cs));
+	ctcopy = strdup(ciphertext);
+	keeptr = ctcopy;
+	ctcopy += FORMAT_TAG_LENGTH;
+	p = strtokm(ctcopy, "$");
+	cs.version = atoi(p);
+	p = strtokm(NULL, "$");
+	cs.algo_id = atoi(p);
+	p = strtokm(NULL, "$");
+	for (i = 0; i < ENC_IV_PREFIX_SIZE; i++)
+		cs.iv[i] =
+		    (atoi16[ARCH_INDEX(p[2 * i])] << 4) | atoi16[ARCH_INDEX(p[2 * i +
+		            1])];
+
+	// Init AES CTR counter
+	cs.iv[ENC_IV_SIZE - 1] = 1;
+
+	p = strtokm(NULL, "$");
+	for (i = 0; i < ENC_SIG_SIZE; i++)
+		cs.encrypted_data[i + 4] =
+		    (atoi16[ARCH_INDEX(p[2 * i])] << 4) | atoi16[ARCH_INDEX(p[2 * i +
+		            1])];
+
+	// Keychain for version 3
+	if (cs.version == 3) {
+		p = strtokm(NULL, "$");
+		for (i = 0; i < ENC_KEYCHAIN_SIZE; i++)
+			cs.keychain[i] =
+			    (atoi16[ARCH_INDEX(p[2 * i])] << 4) | atoi16[ARCH_INDEX(p[2 *
+			            i + 1])];
+	}
+
+	MEM_FREE(keeptr);
+	return (void *)&cs;
+}
+
+static void set_salt(void *salt)
+{
+	cur_salt = (struct custom_salt *)salt;
+}
+
+static int crypt_all(int *pcount, struct db_salt *salt)
+{
+	const int count = *pcount;
+	int index;
+	int nb_keys;
+
+	memset(cracked, 0, sizeof(cracked[0]) * cracked_count);
+	if (1 == cur_salt->version) {
+		nb_keys = 1 << (cur_salt->algo_id - 1);
+	} else {
+		nb_keys = 8;
+	}
+
+#ifdef _OPENMP
+	#pragma omp parallel for
+#endif
+	for (index = 0; index < count; index += MIN_KEYS_PER_CRYPT) {
```

Comment:
Please replace `index += MIN_KEYS_PER_CRYPT` with simply `index++`. The loop body doesn't currently support anything else anyway.

(We use the `index += MIN_KEYS_PER_CRYPT` in some SIMD-capable formats, where that corresponds to how many candidates are processed by one loop iteration when using SIMD.)